nsv13/code/modules/overmap/physics.dm

```diff
@@ -606,3 +615,16 @@ The while loop runs at a programatic level and is thus separated from any thrott
 		proj.fire(theangle)
 		if(speed)
 			proj.set_pixel_speed(speed)
+
+
+/obj/structure/overmap/proc/set_posi2real(x, y) //sets an overmap object to a realworld object and updates the offset based on a given position vector
```

Comment:
This is actually pretty nice!
You could also honestly refactor pixelMovement into a component to be applied onto anything like this..